Sources/SwiftDocC/Indexing/IndexJSON/Index.swift

```diff
@@ -0,0 +1,188 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import SymbolKit
+
+public struct Index: Codable {
```

Comment:
The `Index` vs `NavigatorIndex` naming isn't totally clear to me. Can we use the same model for LMDB and JSON outputs (and abstract the serialisation logic), and if not, why? If `Index` is an implementation detail to serialise things to JSON, maybe we should keep it `internal`.

Reply:
I agree this is confusing. Ideally- I'd like to break things out here so we'd have something like:

1. NavigatorIndex
2. JSONNavigatorIndexReaderWriter
3. LMDBNavigatorIndexReaderWriter

And the JSON representation would be totally inter-changeable with the LMDB representation.

I looked into doing that now but I don't have bandwidth for that full refactor with the way the code is currently architected.

> If Index is an implementation detail to serialise things to JSON, maybe we should keep it internal.

I'm not opposed to that but there is some value in letting clients adopt the API for parsing.